core/templates/components/state-editor/state-hints-editor/state-hints-editor.component.html

```diff
@@ -1,109 +1,139 @@
             <h3 class="oppia-exp-hints-card-header">Hints</h3>
             <i class="fa fa-caret-down"
-               ng-if="!hintCardIsShown"
+               *ngIf="!hintCardIsShown"
                aria-hidden="true">
             </i>
             <i class="fa fa-caret-up"
-               ng-if="hintCardIsShown"
+               *ngIf="hintCardIsShown"
                aria-hidden="true">
             </i>
           </div>
         </div>
-        <div class="state-hints-content-container oppia-mobile-collapsible-card-content" ng-if="hintCardIsShown">
+        <div class="state-hints-content-container oppia-mobile-collapsible-card-content" *ngIf="hintCardIsShown">
           <div class="oppia-add-hint-button-container">
-            <div ng-if="EditabilityService.isEditableOutsideTutorialMode()">
+            <div *ngIf="editabilityService.isEditableOutsideTutorialMode()">
               <button type="button"
                       class="btn btn-primary oppia-add-hint-button protractor-test-oppia-add-hint-button"
-                      ng-click="openAddHintModal()"
-                      ng-disabled="StateHintsService.displayed.length >= 5">
-                <[getHintButtonText()]>
+                      (click)="openAddHintModal()"
+                      [disabled]="stateHintsService.displayed.length >= 5">
+                {{getHintButtonText()}}
               </button>
             </div>
           </div>
-          <div ng-if="StateHintsService.displayed.length > 0">
+          <div *ngIf="stateHintsService.displayed.length > 0">
           <!-- An HTML element marked ui-sortable should contain only one element,
           and this element should have an ng-repeat defined on it. See the
           ui-sortable documentation for more details. -->
-            <ul class="nav oppia-option-list nav-stacked nav-pills" role="tablist"
-                ui-sortable="HINT_LIST_SORTABLE_OPTIONS"
-                ng-model="StateHintsService.displayed">
-              <!-- Note that adding "track by $index" here seems to mess up the final
+            <div cdkDropList (cdkDropListDropped)="drop($event)" class="nav oppia-option-list nav-stacked nav-pills">
+              <!-- Note that adding "track by index" here seems to mess up the final
               index in the stop() event handler. -->
-              <li ng-repeat="hint in StateHintsService.displayed"
-                  ng-class="{'active': StateHintsService.getActiveHintIndex() === $index}"
-                  class="oppia-rule-block oppia-sortable-hint oppia-prevent-selection">
+              <div [cdkDragDisabled]="stateHintsService.getActiveHintIndex() === index ? true : false" cdkDrag *ngFor="let hint of stateHintsService.displayed; let index = index"
+                  ngClass="{'active': stateHintsService.getActiveHintIndex() === index}" class="oppia-rule-block oppia-sortable-hint oppia-prevent-selection">
```

Comment:
Let's add new lines between each attribute within this `div` to make it easier to read.

Reply:
done